### DISCUSSION TITLE
Need to figure out how to change IRC messages depending on PubSub events

### DISCUSSION THREAD

**Discussion by karl-police** [Chatterino/chatterino2#3499] [Q&A]
This is an example Automod captured Twitch message with a monitored message:

![](https://i.imgur.com/sjQxUxI.png)

The Automod message is a PubSub message and so is the low-trust-users one, the one that says Suspicious User.

Twitch took the "low-trust-users" message and put it within the Automod message, making it clear that the suspicious user message, somehow gets in the original message.

Trying to add "low-trust-users" to the PubSub in Chatterino works, but there are some issues that I wonder on how I should deal with these:

**Just note these only occur for "Monitored Messages" anything else doesn't cause this problem.**

![](https://i.imgur.com/HAJU0vl.png)

The reason this happens is because the message also gets sent to IRC.

&nbsp;

Here is the other issue:

![](https://i.imgur.com/JzUYoxL.png)

And that's all basically.

I was wondering if I could collect ideas on how to deal with this issue.

My thought was that I would look on how deleting messages works on Chatterino, to find the MessageID and "reconstruct the message".

**Comment by karl-police** [comment]
I think I found an idea:

https://github.com/Chatterino/chatterino2/blob/master/src/common/Channel.cpp#L241

What delete does is, it changes the message flag to "Disabled". What I wonder, is if "None" would do the thing, but at the same time, I think the messages for the CTRL + F are separated? Not sure.

Will try out now.

**Reply by karl-police** [reply]
> I think I found an idea:
> 
> https://github.com/Chatterino/chatterino2/blob/master/src/common/Channel.cpp#L241
> 
> What delete does is, it changes the message flag to "Disabled". What I wonder, is if "None" would do the thing, but at the same time, I think the messages for the CTRL + F are separated? Not sure.
> 
> Will try out now.

I couldn't figure out for now.

**Comment by zneix** [comment]
"PubSub changed the way how the chat works" ...how did it actually? I don't get the point of the issue/discussion.

**Reply by karl-police** [reply]
> "PubSub changed the way how the chat works" ...how did it actually? I don't get the point of the issue/discussion.

it's not IRC anymore that has flags like first-message, now it's also PubSub,

so what lowtrustusers does on Twitch, is that it modified the original IRC message by adding like, that "Suspicious User" and all to it. And, we would have to do the same, thing is PubSub sometimes is earlier than IRC, so I think a separate queue has to be made for PubSub messages to match it with IRC messages.

these are some example responses:
https://paste.ivr.fi/quxureziqe.swift
https://paste.ivr.fi/nabikevoqu.swift

**Reply by LosFarmosCTL** [reply]
There is already some logic in place to replace TO/ban messages with their PubSub equivalent (See i.e. #2804), although as you can see from that issue its definitely far from perfect.

Might even have something to do with a PubSub message being received before the IRC message and chatterino not knowing what message to replace? I havent looked at that at all though, so just guessing here...

**Reply by karl-police** [reply]
> There is already some logic in place to replace TO/ban messages with their PubSub equivalent (See i.e. #2804), although as you can see from that issue its definitely far from perfect.
> 
> Might even have something to do with a PubSub message being received before the IRC message and chatterino not knowing what message to replace? I havent looked at that at all though, so just guessing here...

Yeah, I don't know how Twitch does it, but you could check it, I did too but it's not really readable.

But my idea would be to put the messages in a "PubSub replace message queue" and check for the message ID in the IRC, to replace it with, I guess.
